### PR TITLE
Fix double refreshes in workspace tabs (related to GAWB-2791)

### PR DESCRIPTION
**Pull request by MattPutnam**
A bunch of minor cleanup too, the important bits are part of the first commit, and the last commit.

There were double refreshes happening in the method configs and monitor tabs. Method configs had an extraneous request-refresh doing all sorts of bad things, and both tabs had a problem where if you were drilled inside the tabs (editing a method config, viewing a submission) and navigated to another tab it would double-refresh.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/details.cljs [hunk 1]
```diff
@@ -139,27 +139,28 @@
                              (utils/restructure workspace-id workspace workspace-error request-refresh))])
                ANALYSIS (react/create-element
                          [analysis-tab/Page {:ref ANALYSIS :workspace-id workspace-id}])
+               NOTEBOOKS (react/create-element
+                          [notebooks-tab/Page
+                           (merge {:ref NOTEBOOKS}
+                                  (utils/restructure workspace-id workspace))])
```
Comment (davidangb) [inline src/cljs/main/broadfcui/page/workspace/details.cljs:145]: did this move for a reason?

**Review by davidangb — APPROVED** [review]
:+1: looks good, one question inline that is not a blocker.

File: src/cljs/main/broadfcui/page/workspace/details.cljs (hunk 1 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/page/workspace/details.cljs:145]: I moved it to match the display order of the tabs.

File: src/cljs/main/broadfcui/page/workspace/details.cljs [hunk 2]
```diff
@@ -139,27 +139,28 @@
...
-                                  (utils/restructure workspace-id workspace))]))))]]))
+                                (select-keys props [:submission-id :workflow-id]))]))))]]))
    :component-will-mount
    (fn [{:keys [this]}]
      (this :-refresh-workspace))
    :component-will-receive-props
-   (fn [{:keys [this after-update]}]
-     (after-update this :-refresh-workspace))
+   (fn [{:keys [props next-props this after-update]}]
+     (when (not= (:tab-name props) (:tab-name next-props))
```
Comment (aednichols) [inline src/cljs/main/broadfcui/page/workspace/details.cljs:162]: Cool! I keep learning new useful things one can plug into this method.

**Review by aednichols — APPROVED** [review]
👍

**Pull request merged**